File: web/src/types/game.ts
import { Player } from "./player"

export type GameStatus = "in_progress" | "draw" | "x_winner" | "o_winner" | "withdraw" | "searching"

export interface Move {
	playerId: string
	position: number
}

export interface Game {
	id: string
	playerX?: Player
	playerO?: Player
	status: GameStatus
	started_at: string
	moves: Move[]
}

File: app/app/src/main/java/com/example/app/data/Game.kt
package com.example.app.data

object GameStatus {
    const val IN_PROGRESS = "in_progress"
    const val DRAW = "draw"
    const val X_WINNER = "x_winner"
    const val O_WINNER = "o_winner"
    const val WITHDRAW = "withdraw"
    const val SEARCHING = "searching"
}

data class Move(
    val playerId: String? = null,
    val position: Int? = null
)

data class Game(
    val id: String? = null,
    val playerX: Player? = null,
    val playerO: Player? = null,
    val status: String? = null,
    val moves: List<Move>? = null
)

data class GameCreation(
    val playerX: Player,
    val status: String,
    val moves: List<Move>
)